File: server/src/main/java/ru/fullrest/mfr/server/configuration/CacheConfiguration.java
package ru.fullrest.mfr.server.configuration;

import com.github.benmanes.caffeine.cache.Caffeine;
import com.github.benmanes.caffeine.cache.Ticker;
import org.springframework.cache.Cache;
import org.springframework.cache.CacheManager;
import org.springframework.cache.annotation.EnableCaching;
import org.springframework.cache.caffeine.CaffeineCache;
import org.springframework.cache.support.SimpleCacheManager;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.ArrayList;
import java.util.concurrent.TimeUnit;

@Configuration
@EnableCaching
public class CacheConfiguration {

    public static final String PROPERTY_CACHE = "PropertyCache";
    public static final String USER_CACHE = "UserCache";
    public static final String UPDATE_CACHE = "UpdateCache";
    public static final String STATISTICS_CACHE = "StatisticsCache";

    @Bean
    public Ticker ticker() {
        return Ticker.systemTicker();
    }

    @Bean
    public CacheManager cacheManager(Ticker ticker) {
        ArrayList<Cache> caches = new ArrayList<>();
        caches.add( new CaffeineCache(
                PROPERTY_CACHE,
                Caffeine.newBuilder()
                        .expireAfterWrite(1, TimeUnit.DAYS)
                        .ticker(ticker)
                        .build()
        ));
        caches.add( new CaffeineCache(
                USER_CACHE,
                Caffeine.newBuilder()
                        .expireAfterWrite(1, TimeUnit.DAYS)
                        .ticker(ticker)
                        .build()
        ));
        caches.add( new CaffeineCache(
                UPDATE_CACHE,
                Caffeine.newBuilder()
                        .expireAfterWrite(1, TimeUnit.DAYS)
                        .ticker(ticker)
                        .build()
        ));
        caches.add( new CaffeineCache(
                STATISTICS_CACHE,
                Caffeine.newBuilder()
                        .expireAfterWrite(10, TimeUnit.MINUTES)
                        .ticker(ticker)
                        .build()
        ));

        SimpleCacheManager cacheManager = new SimpleCacheManager();
        cacheManager.setCaches(caches);
        return cacheManager;
    }
}

File: server/manager/src/main/kotlin/com/lezenford/mfr/manager/configuration/CacheConfiguration.kt
package com.lezenford.mfr.manager.configuration

import com.github.benmanes.caffeine.cache.Caffeine
import com.github.benmanes.caffeine.cache.Ticker
import org.springframework.cache.CacheManager
import org.springframework.cache.annotation.EnableCaching
import org.springframework.cache.caffeine.CaffeineCache
import org.springframework.cache.support.SimpleCacheManager
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.util.concurrent.TimeUnit

@EnableCaching
@Configuration
class CacheConfiguration {

    @Bean
    fun ticker(): Ticker = Ticker.systemTicker()

    @Bean
    fun cacheManager(ticker: Ticker): CacheManager =
        SimpleCacheManager().also {
            it.setCaches(
                listOf(
                    CaffeineCache(
                        USER_CACHE,
                        Caffeine.newBuilder()
                            .expireAfterAccess(1, TimeUnit.DAYS)
                            .ticker(ticker)
                            .build()
                    ),
                    CaffeineCache(
                        MODULE_CACHE,
                        Caffeine.newBuilder()
                            .expireAfterAccess(30, TimeUnit.DAYS)
                            .ticker(ticker)
                            .build()
                    ),
                    CaffeineCache(
                        JWT_CACHE,
                        Caffeine.newBuilder()
                            .expireAfterWrite(50, TimeUnit.MINUTES)
                            .ticker(ticker)
                            .build()
                    )
                )
            )
        }
}

const val USER_CACHE = "UserCache"
const val MODULE_CACHE = "ModuleCache"
const val JWT_CACHE = "JwtCache"